javalib/src/main/scala/java/math/BigInteger.scala

```diff
@@ -0,0 +1,1500 @@
+   * @throws NullPointerException if {@code val == null}.
+   * @throws NumberFormatException if {@code val} is not a valid representation
+   *           of a {@code BigInteger} or if {@code radix < Character.MIN_RADIX}
+   *           or {@code radix > Character.MAX_RADIX}.
+   */
+  def this(s: String, radix: Int) {
+    this()
+    checkNotNull(s)
+    if ((radix < java.lang.Character.MIN_RADIX) || (radix > java.lang.Character.MAX_RADIX)) {
+      throw new NumberFormatException("Radix out of range")
+    }
+    if (s.isEmpty) {
+      throw new NumberFormatException("Zero length BigInteger")
+    }
+    this.setFromString(s, radix)
+  }
+
+  /**
+   * Constructs a new {@code BigInteger} instance from the string
+   * representation. The string representation consists of an optional minus
+   * sign followed by a non-empty sequence of decimal digits.
+   *
+   * @param s string representation of the new {@code BigInteger}.
+   * @throws NullPointerException if {@code val == null}.
+   * @throws NumberFormatException if {@code val} is not a valid representation
+   *           of a {@code BigInteger}.
+   */
+  def this(s: String) {
+    this(s, 10)
+  }
+
+  /**
+   * Constructs a number which array is of size 1.
+   *
+   * @param sign the sign of the number
+   * @param value the only one digit of array
+   */
+  def this(sign: Int, value: Int) {
+    this()
+    this.sign = sign
+    numberLength = 1
+    digits = Array(value)
+  }
+  
+  /**
+   * Creates a new {@code BigInteger} with the given sign and magnitude. This
+   * constructor does not create a copy, so any changes to the reference will
+   * affect the new number.
+   *
+   * @param signum The sign of the number represented by {@code digits}
+   * @param digits The magnitude of the number
+   */
+  def this(signum: Int, digits: Array[Int]) {
+    this()
+    if (digits.length == 0) {
+      this.sign = 0
+      this.numberLength = 1
+      this.digits = Array(0)
+    } else {
+      this.sign = signum
+      this.numberLength = digits.length
+      this.digits = digits
+      this.cutOffLeadingZeroes()
+    }
+  }
+
+  /**
+   * Constructs a number without to create new space. This construct should be
+   * used only if the three fields of representation are known.
+   *
+   * @param sign the sign of the number
+   * @param numberLength the length of the internal array
+   * @param digits a reference of some array created before
+   */
+  def this(sign: Int, numberLength: Int, digits: Array[Int]) {
+    this()
+    this.sign = sign
+    this.numberLength = numberLength
+    this.digits = digits
+  }
+
+  /**
+   * Creates a new {@code BigInteger} whose value is equal to the specified
+   * {@code long}.
+   *
+   * @param sign the sign of the number
+   * @param val the value of the new {@code BigInteger}.
+   */
+  def this(sign: Int, lVal: Long) {
+    this()
+    this.sign = sign
+    if ((lVal & 0xFFFFFFFF00000000L) == 0) {
```

Comment:
First extract
```scala
val hi = (lval >>> 32).toInt
```
then compare `hi == 0`. Reuse `hi` in the `else` branch. This will create less RuntimeLongs.